Firestore/Source/Local/FSTQueryData.mm

```diff
@@ -74,13 +74,15 @@ - (BOOL)isEqual:(id)object {
 
   FSTQueryData *other = (FSTQueryData *)object;
   return [self.query isEqual:other.query] && self.targetID == other.targetID &&
-         self.purpose == other.purpose && self.snapshotVersion == other.snapshotVersion &&
+         self.sequenceNumber == other.sequenceNumber && self.purpose == other.purpose &&
+         self.snapshotVersion == other.snapshotVersion &&
          [self.resumeToken isEqual:other.resumeToken];
 }
 
 - (NSUInteger)hash {
   NSUInteger result = [self.query hash];
```

Comment:
This can use our [`Hash` utility ](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/util/hashing.h)

```c++
return util::Hash([self.query hash], self.targetID, self.sequenceNumber, self.purpose,
                  self.snapshotVersion, [self.resumeToken hash]);
```

Reply:
Done.